Flows for Q&A

smcnally opened the pull request.

smcnally commented:
Thanks, again, for making chat-bubble available. 
Make or request changes as you see fit.

dmitrizzle commented on examples/6-advanced-flows.html:
```diff
@@ -0,0 +1,213 @@
+            },
+            {
+                "question": "Yes",
+                "answer": "externalResourceFunction"
```
I think this function should also give user an option to continue chat. Otherwise the window will pop up and there will be no way to continue from here on.

smcnally replied:
Will add that.

smcnally replied:
I've addressed all but this one and can use guidance re getting start over / ice to work within the script.

dmitrizzle commented on examples/6-advanced-flows.html:
```diff
@@ -0,0 +1,213 @@
+
+    // this function is called when user asks for banana
+    externalResourceFunction = function () {
+        window.open("https://meanbusiness.com/gamepath/coralproject/coral-swiper-180306/","_blank",
```
What is this website?

smcnally replied:
[As below](https://github.com/dmitrizzle/chat-bubble/pull/53#discussion_r179615446), this is another example use case I did for the Coral Project to represent personality types in News communities. Will update this link.

dmitrizzle commented on examples/6-advanced-flows.html:
```diff
@@ -0,0 +1,213 @@
+        // "says" defines an array of sequential bubbles
+        // that the bot will produce
+        says: [
+            "Hello",
```
The way the users will see this example is by clicking the link without any of the comments. I think it's important to let them know that this is an example chat flow that they could explore some advanced features from.

smcnally replied:
I'll update accordingly.

dmitrizzle commented on examples/6-advanced-flows.html:
```diff
@@ -0,0 +1,213 @@
+    answer: "chapter-one" // key for the next conversation object
+            },
+            {
+        question: "No. Thanks [Escape Valve hit]", // label for the reply option
```
Not sure if the information in the square brackets is helpful in this chat. Perhaps we can explain in plain English what the bot is doing?

smcnally replied:
Understood. As you note above, the majority of people will experience the example front-end first, rather than via code comments. I'll update accordingly.

dmitrizzle commented on examples/6-advanced-flows.html:
```diff
@@ -0,0 +1,213 @@
+            "[Questions | Info | Requests for feedback]",
+        ],
+        "reply": [
+            {
```
This set (and the following few) of answers is a little confusing. I understand this is an example, but perhaps we could make up some imaginary questions and answers to match closer to an actual use-case?

smcnally replied:
I can keep the narrative to Bananas and Ice Cream for continuity with your other examples. 

The chat is based on [an example use case](https://meanbusiness.com/gamepath/chat-bubble/examples/Coral-intro-chat.html) I did for the [Coral Project](https://www.coralproject.net/) to represent personality types in News communities.

dmitrizzle replied:
This is awesome, love the project! How about using the bananas & ice cream for now; but would you like to add a link to your project to the main Readme.md?

smcnally commented:
All makes sense. Acknowledging here. Will update.

smcnally commented:
I've  kept the narrative to Bananas and Ice Cream, added the flow comments on the front end, removed the square brackets. All but [adding Start Over](https://github.com/dmitrizzle/chat-bubble/pull/53#discussion_r179360302) to the script should be handled. Please say the word if you'd like more changes.

dmitrizzle commented:
Hi @smcnally thank you for the changes!
I've edited your example a bit for clarity and have fixed the external function so that it does not interrupt the chat. If this looks good to you, I'll add more detailed comments in to the code and merge.

dmitrizzle commented:
Note to self: should add dimensions to GIFs to ensure that they do not jump when loaded.
